File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.c
int longestOnes(int* nums, int nums_size, int k){
    int l, r, max_seq, seq;
    l = r = max_seq = 0;
    
    while (r < nums_size) {
        if (nums[r++] == 0)
            k--;
        if (k < 0 && nums[l++] == 0)
            k++;
        seq = r - l;
        max_seq = (max_seq >= seq) ? max_seq : seq;
    }
    return max_seq;
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.go
func longestOnes(nums []int, k int) int {
    seq := 0
    maxSeq := 0
    
    for i, _ := range nums {
        seq = calculateSeq(nums[i:], k)
        maxSeq = max(maxSeq, seq)
    }
    return maxSeq
}

func calculateSeq(nums []int, k int) int {
    idx := 0
    for k != 0 && idx < len(nums) {
        if nums[idx] == 0 {
            k--
        }
        idx++
    }
    for idx < len(nums) {
        if nums[idx] == 0 {
            break
        }
        idx++
    }
    return idx
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}